worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionDataWriter.java

```diff
@@ -293,14 +293,15 @@ public void flush(boolean finalFlush, boolean fromEvict) throws IOException {
           // read flush buffer to generate correct chunk offsets
           // data header layout (mapId, attemptId, nextBatchId, length)
           if (numBytes > chunkSize) {
-            ByteBuffer headerBuf = ByteBuffer.allocate(16);
+            ByteBuffer headerBuf = ByteBuffer.allocate(PushDataHeaderUtils.BATCH_HEADER_SIZE);
```

Comment:
If a worker receives a data buffer from the an old version client or not supported engine client, will this modification yield accurate results?

Reply:
If a worker receives a data buffer from an old version client, we will check the checksum flag in the highest bit of batchId for compatibility. Similarly, if the data comes from an unsupported engine's client but follows the old version data header layout, it will also be compatible.